gramps/gen/lib/serialize.py

```diff
@@ -132,7 +132,7 @@ def to_dict(obj):
     :returns: A dictionary.
     :rtype: dict
     """
-    return json.loads(to_json(obj))
+    return DataDict(json.loads(to_json(obj)))
```

Comment:
Given you have `obj` already, an optimisation would be to use it set to `DataDict._object` during init

In fact do you even need to initialise the `DataDict` super class `dict` if _object is set? 
Just leave the dict empty, except for `_object`
I think `DataDict.__getattr__` will end up calling `return getattr(self["_object"], key)`.

If the above is right, it avoids the `json.loads(to_json())` inefficiency

Reply:
A `DataDict` then has 3 possible states:
1. just the object attributes in the dict
2. the object attributes in the dict _and_ the true object in `dict["_object"]`
3. just the true object in `dict["_object"]`

Reply:
Interesting! I didn't about defaulting to the _object attributes. Very clever!

Let me see if I can create an interface for that...

Reply:
Actually, since `DataDict` mirrors the object now, we can just return the object, right?

Changing:
```
return DataDict(json.loads(to_json(obj)))
```
to just:
```
return obj
```

Reply:
(It would be the wrong type, but would function correctly)

Reply:
well it's the pythonic way, even if my C++ soul is crying 😉
If we accept that, you don't need to call `to_dict` at all, since it just returns it's argument, unaltered.

Reply:
I think I'll make an DataDict wrapper for objects.

Reply:
Added `DataDict(obj)` in #1839

Reply:
I'll update this code after #1839 is merged.